### ISSUE TITLE
Photo section load endless

### ISSUE THREAD

**Issue by CordlessWool** [nextcloud/android#1682]
### Actual behaviour
photo section is loading endless. Monitoring Server show that the cpu use increase for an minute but the the app do not stop loading. thumbnails in "all files"-section are shown normal

### Expected behaviour
should show photos after created thumbnails on server
 
### Steps to reproduce
1. open app
2. click to photo seciton
3. 

### Environment data
Android version: 7.1.2

Device model: Nexus 5x with LineageOS

Stock or customized system: LineageOS

Nextcloud app version: 2.0.0

Nextcloud server version: 12.0.3

#### Web server error log
no specific error on server log, only Sabre\DAV\Exception\BadRequest: expected filesize 1343225856 got 57344 at same day but not same time

#### Nextcloud log (data/nextcloud.log)
```
Insert your Nextcloud log here
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by phatair** [comment]
Same problem here. 
Galaxy s7 edge with android 7.0
Nextcloud app 2.0.0
Nextcloud server 12.0.3

Photo section and the search is loading endless. 
I didn't checked the server logs. I will do this tomorrow.

**Comment by AndyScherzinger** [comment]
The photo search might take quite a while at this time since it fetches a search result with all images-search-items from the server and then renders a result list this might, depending on the image count take ~15 minutes for 5000+ images (like on my server).
This will get better as soon as:

* the app moves to recycler view for lists
* fetches the result in blocks while it still be just as fast as the server does the search...

**Comment by memen45** [comment]
Same here. Isn't it possible to limit search to specific directories? For example 'Photos' or 'Pictures' would be directories you want to load, but 'Backup/Photos' probably not.

**Comment by phatair** [comment]
I dont have a lot of pictures on my cloud - around 300 and I don't see any pictures in the photo section - Only "loading..."

**Comment by CordlessWool** [comment]
I do not know how much pictures I have exactly but I open the app and let it work for a while (much more then 15 minutes). Is it a problem if screen goes black, I am not 100% familiar with android? But also the cpu usage on server goes back to normal after around 1 minute

**Comment by AndyScherzinger** [comment]
@CordlessWool yeah, most of the processing takes place on the client due to a not yet optimal implementation (ListView instead of RecyclerView, reading all data at the same time instad on endless scrolling/block-reading).

**Comment by phatair** [comment]
I don't get this running in my phone. 
is there any release date for a fix?

**Comment by influenist** [comment]
I sort all pictures based on year / month. With around 300 GB of pictures and 500 a month. The app is sooooo slow a honey. Even the web is loads bad with the Preview app enabled. Basically it doesn't fix anything. Compared to Dropbox, this solution is very FAR behind... 

I run NC on 6 core with super fast SSDs, 4 GB ram and all optimisations which you can imagine like NGINX, APU, Redis.

**Comment by KeesKemper** [comment]
I've put in all directories .nomedia to prevent that the pictures are displayed. Even then the phone keeps saying loading when the photo's option is selected. Is there any sight on a fix?

**Comment by tobiasKaminsky** [comment]
There are several things which will improve speed: #2087, https://github.com/nextcloud/server/pull/8023 and we want to show only the first 100 (to be discussed) images and when reaching the bottom of the list, it should automatically append the next 100.

**Comment by rullzer** [comment]
As pointed out by @tobiasKaminsky we are working on improving this. The big issue is that server side support is needed. And https://github.com/nextcloud/server/pull/8023 is a big step in that direction. This will allow the client to do better queries resulting in faster and smaller responses.

Currently the only thing the client can do is say: "Give me all the pictures!" which on most setups will take a significant amount of time to fetch/process and return.

So long story short we are working on this and hopefully it will all run as smooth as possible on NC14

**Comment by phatair** [comment]
Ok, i hope that will be fixed in NC14 :)
I'm running now NC13 and the newest android app and its still the same problem.

**Comment by suokunlong** [comment]
Is this the same issue as in issue  #2850?

**Comment by elpedriyo** [comment]
No fix yet on displaying photos quicker on the android app?

**Comment by AndyScherzinger** [comment]
The server doesn't support reading the photos in blocks, thus this can't be implemented at the moment.

**Comment by mlaverdiere** [comment]
One relatively light workaround I see at the development level could be that, when the user select the photo/gallery section in the Android app, instead of trying to show all the pictures available in all the folders (which never succeeds with medium/large collection), it could  behave like the NC web gallery app, i.e. only showing the first level folders where pictures are available. Then, navigating in each of these folders, where presumably there is less pictures to show, could give a chance to the app. to actually show something!

This is not the perfect solution of course but it could make the photo section in the Android app usable (right now, it's pretty useless).

That said, the ultimate goal should still be to be able to show all the pictures in the photo section, with a reasonable delay, like Dropbox and other comparable apps do.

**Comment by s6410** [comment]
It works fine on iPhone, where all photos are listed as expected, but doesn't work on Samsung S8, where I only get the spinning wheel. 

Nextcloud server: 15.0.5
Nextcloud client: 3.5.1
Phone: Samsung S8 Android 9

**Comment by rhatguy** [comment]
Whats the latest on this issue?  I've been avoiding moving my large photo collection into Nextcloud as the android app is completely unusable with a large collection.  Are changes still being worked on on the server side to speed up the android client loading images?

**Comment by s6410** [comment]
Still behaving the same on my Samsung S8. My server is the latest version as well as the app. No change:(

**Comment by tobiasKaminsky** [comment]
There is now a PR ready to test:
https://github.com/nextcloud/android/pull/2250#issuecomment-512712749

**Comment by tobiasKaminsky** [comment]
To centralize this please report anything found in #2250.

**Comment by KlavsKlavsen** [comment]
I just installed nextcloud (running as docker instance - with haproxy in front) - and it "seems to work fine". I connected the fdroid app on my lineageOS 15.1 to it - and I uploaded the 4 images I have (only 4). They now shop under InstantlyUPloaded/Camera - and show in webbrowser as well.
When I select "photos" - it shows the 4 images - but even though everything seems to work - the wheel keeps spinning in the app - below the pictures.. This can't be it waiting for some request and never timing out?
